libraries/chain/block_state.cpp

```diff
@@ -51,6 +64,19 @@ block_state::block_state(const block_state_legacy& bsp) {
    core = finality_core::create_core_for_genesis_block(bsp.block_num()); // [if todo] instant transition is not acceptable
    activated_protocol_features = bsp.activated_protocol_features;
 
+   // construct valid structure
+   valid = valid_t {
+      .finality_merkel_tree = incremental_merkle_tree{},
+      .finality_mroots      = { digest_type{} }, // for genesis block itself
+      .last_final_block_num = bsp.block_num()
+   };
+   valid_t::finality_leaf_node_t node {
```

Comment:
I think we also discussed adding a "finality leaf version number" here, which will help proper handling of different versions moving forward.

Reply:
Does the `light_header_protocol_version` in `finality_digest_data_v0_t` of block_header_state.cpp serve that purpose?

Reply:
No, this here is a different version number, as it applies to the "dynamic data" portion of the leaf data. It will eventually be used when we incorporate state proofs at some point in the future. In the diagram, I called it "finality_tree_leaf_version".